src/core/agent/services/connectionService.ts

```diff
@@ -261,31 +260,57 @@ class ConnectionService extends AgentService {
     connectionId: string,
     note: ConnectionNoteProps
   ): Promise<void> {
-    await this.connectionNoteStorage.save({
-      id: uuidv4(),
-      title: note.title,
-      message: note.message,
-      connectionId,
+    const id = new Salter({}).qb64;
+    await this.props.signifyClient.contacts().update(connectionId, {
+      [`${ConnectionService.CONNECTION_NOTE_PREFIX_KEY}${id}`]: JSON.stringify({
+        ...note,
+        id: `${ConnectionService.CONNECTION_NOTE_PREFIX_KEY}${id}`,
+      }),
     });
   }
 
   async updateConnectionNoteById(
+    connectionId: string,
     connectionNoteId: string,
     note: ConnectionNoteProps
   ) {
-    const noteRecord = await this.connectionNoteStorage.findById(
-      connectionNoteId
-    );
-    if (!noteRecord) {
-      throw new Error(ConnectionService.CONNECTION_NOTE_RECORD_NOT_FOUND);
-    }
-    noteRecord.title = note.title;
-    noteRecord.message = note.message;
-    await this.connectionNoteStorage.update(noteRecord);
+    const connection = await this.props.signifyClient
+      .contacts()
+      .get(connectionId)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (/404/gi.test(status)) {
+          throw new Error(`${Agent.MISSING_DATA_ON_KERIA}: ${connectionId}`);
+        } else {
+          throw error;
+        }
+      });
+    const connectionNote = JSON.parse(connection[connectionNoteId]);
+    connectionNote.title = note.title;
+    connectionNote.message = note.message;
+    await this.props.signifyClient.contacts().update(connectionId, {
```

Comment:
Unless there's a bug, `.update` should keep the old data there. So you should just be able to directly call this without `contacts().get`